test/test.js

```diff
@@ -99,99 +51,89 @@ describe('Azure Swagger Schema Validation', function () {
+                  if (error) {
+                    messages.push('An error occurred while accessing the swagger doc ' +
+                      docUrl + ' from the documents list. The error is ' + util.inspect(error, { depth: null }));
+                  }
+                  if (response.statusCode !== 200) {
+                    messages.push('\'' + response.statusCode + '\': \'File Not Found\'- error occurred while accessing the swagger doc ' +
+                      docUrl + ' from the documents list.');
+                  }
+                  loopCallback();
+                });
+              } else {
+                //check whether the file exists
+                if (!fs.existsSync(docUrl)) {
+                  messages.push('\'File Not Found\': error occurred while accessing the swagger doc ' +
+                    docUrl + ' from the documents list on the host filesystem.');
                 }
                 loopCallback();
-              });
-            } else {
-              //check whether the file exists
-              if (!fs.existsSync(docUrl)) {
-                messages.push('\'File Not Found\': error occurred while accessing the swagger doc ' +
-                  docUrl + ' from the documents list on the host filesystem.');
               }
-              loopCallback();
-            }
-          }, function (err) {
-            if (err) {
-              throw err;
-            }
-            if (messages.length > 0) {
-              throw new Error(JSON.stringify(messages));
-            }
+            }, function (err) {
+              if (err) {
+                throw err;
+              }
+              if (messages.length > 0) {
+                throw new Error(JSON.stringify(messages));
+              }
+              done();
+            });
+          } else {
             done();
-          });
-        } else {
-          done();
-        }
+          }
+        });
       });
-    });
-  }).value();
-
-  _(examples).each(function (example) {
-    it('x-ms-examples: ' + example + ' should be a valid x-ms-example.', function (done) {
-      fs.readFile(example, 'utf8', function (err, data) {
-        if (err) { done(err); }
-        var parsedData;
-        try {
-          parsedData = JSON.parse(stripBOM(data));
-        } catch (err) {
-          throw new Error("example file " + example + " is an invalid JSON. " + util.inspect(err, { depth: null }));
-        }
+    }).value();
+  });
 
-        var valid = validator.validate(parsedData, exampleSchema);
-        if (!valid) {
-          var error = validator.getLastErrors();
-          throw new Error("Schema validation failed: " + util.inspect(error, { depth: null }));
-        }
-        assert(valid === true);
-        done();
+  describe('Azure x-ms-example schema validation:', function () {
+    _(utils.examples).each(function (example) {
+      it('x-ms-examples: ' + example + ' should be a valid x-ms-example.', function (done) {
```

Comment:
This is schema validation for x-ms-examples. This will run only if x-ms-examples is present.